### ISSUE TITLE
Produce more helpful build errors

### ISSUE THREAD

**Issue by cassiersg** [mcarton/rust-derivative#52]
For the code:
```rust
#[derive(Debug, Clone, Derivative)]                                                                                                                                                                                
#[derivative(                                                                                                                                                                                                      
    PartialEq = "feature_allow_slow_enum",                                                                                                                                                                         
    PartialOrd = "feature_allow_slow_enum",                                                                                                                                                                        
    Ord = "feature_allow_slow_enum",                                                                                                                                                                               
    Eq = "feature_allow_slow_enum"                                                                                                                                                                                               
)] 
pub enum Foo {
    // ...
}
```
the compiler error message is:
```
error: proc-macro derive panicked                                                                                                                                                                                  
  --> example.rs:23:24                                                                                                                                                                              
   |                                                                                                                                                                                                               
23 | #[derive(Debug, Clone, Derivative)]                                                                                                                                                                           
   |                        ^^^^^^^^^^                                                                                                                                                                             
   |                                                                                                                                                                                                               
   = help: message: unknown attribute  
```
This error message is not very helpful (compiling with the `--verbose` flag does not give more immediately exploitable information).
It would be better to have an error message such as `Attribute feature_allow_slow_enum unsuported for trait Eq`.

**Version**

```
rustup 1.19.0 (2af131cf9 2019-09-08)                                                                                                                                                                                                                                                                                                                                                        
rustc 1.37.0 (eae3437df 2019-08-13)                                                                                                                                                                                
cargo 1.37.0 (9edd08916 2019-08-02)  
```

 - Version of `derivative`: git 5ae882d987665bdb29105d8663ca17a70446950b

**Comment by ionut-arm** [comment]
I second this. It seems to be a problem with deriving for enums. My use case (see below) gets the same `proc-macro derive panicked` because of `unknown attribute`. The exact same attributes work just fine on a struct.
```Rust
#[derive(Derivative)]
#[derivative(Debug)]
pub enum Ipc {
    /// Communication using Unix domain sockets
    UnixSocket {
        /// Path at which the socket can be found
        path: PathBuf,
        /// Timeout for reads and writes on the streams
        timeout: Option<Duration>,
    },
    #[cfg(test)]
    #[derivative(Debug = "ignore")]
    MockIpc(SyncMockStream),
}
```

**Comment by mcarton** [comment]
This was solved in 2.1.1, eg. the example above now reports

```rust
error: Unknown attribute `ignore` for trait `Debug`
  --> src/main.rs:15:26
   |
15 |     #[derivative(Debug = "ignore")]
   |                          ^^^^^^^^
```
